### PR TITLE
Add static representation of configuration options

### PR DESCRIPTION
**Pull request by shonfeder**
Contributes towards #1174

## Context

This adds traits with values to represent all the options that we are
currently setting in the `at.forsyte.apalache.tla.tooling.opt.Cmd*`
classes. None of those options are actually transferred over here.

The proposed design uses traits to specify all the options, with the idea that
various subprograms can depend on mixins of the traits to statically declare the
configurations they respond to.

## Reviewing

This changeset just prepares the representation of the options, with the aim of
obtaining feedback and clearing the way for the changeogver. The followup PR
will remove the `PassOptions` class, replacing it with objects that implement
the configuration traits needed by the respective subcommands.

- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Rebase once #2023 is merged in

### CODE REVIEW

**Comment by shonfeder** [comment]
Note this PR is quite small, and reviewing should hold off until the pending PRs it branches off of are merged in and this branch is rebased.

File: mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala
Comment (shonfeder) [inline mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala:30]: This was just in the wrong place.

File: mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala
Comment (shonfeder) [inline mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala:4]: Realized we can declare the package just by putting it in the file and giving the package name, so no need for the object.

**Review by shonfeder — COMMENTED** [review]
@thpani this is ready for review now!

I suggest setting the diff to ignore whitespace, as it cuts the diff down to just about 60 lines.

**Review by thpani — APPROVED** [review]
LGTM

**Pull request merged**
